doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+        AppCatalog catalog{ co_await appInstaller.GetAppCatalogAsync(packageSource) };
+        co_await catalog.OpenAsync();
+        co_return catalog;
+    }
+```
+
+# 4 Remarks
+
+Notes have been added inline throughout the api details.
+
+The biggest open question right now is whether the api needs to provide a way for the client to verify that the server is actually the Windows Package Manager package.
+The concern is the following scenario: 
+The client app has packageManagement capability which currently only allows install of msix applications.
+The user clicks on buttons in the client app to start an install of Microsoft.VSCode. The client app calls the server to do the install, and a UAC prompt is shown.
+The UAC prompt indicates that the app asking for permission is not Microsoft.VSCode, but rather some other app, however the user does not bother to read the prompt and simply clicks accept.
+In this way, an app that takes over the com registration of the Windows Package Manager could use the user's mistake to allow itself to escalate.
+A proposed mitigation is provided in OpenCatalogAsync which is an optional call.
+
+Another question is whether it's required to provide a wrapper api for this api to avoid callers having to use CoCreateInstance, or write their own wrapper to project into c#.
+We could provide a 1 to 1 wrapper implementation in Microsoft.Management.Deployment.Client.
+
+Naming of the items is as always a tricky question. For this api there are multiple similar apis that are relevant with regard to naming and consistency. There is the
+Windows Package Manager command line which uses "source" to describe the various repositories that can host packages and "search" to describe looking up an app. 
+https://docs.microsoft.com/en-us/windows/package-manager/winget/
+There is the Windows::ApplicationModel::PackageCatalog which exists as a Windows API for installing packages and monitoring their installation progress.
+https://docs.microsoft.com/en-us/uwp/api/windows.applicationmodel.packagecatalog?view=winrt-19041
+And there is Windows.Management.Deployment.PackageManager which allows packages with the packageManagement capability to install msix apps and uses "Find" to describe looking up an app
+https://docs.microsoft.com/en-us/uwp/api/windows.management.deployment.packagemanager?view=winrt-19041 
+
+I've chosen to align with the Windows APIs in using *Catalog and Find. But this may cause some confusion if callers are using both the winget command line and this API.
+In particular a problem may be that since this API does not yet propose to implement adding a "source", client applications would need to work with the command line
+interface in order to do that which may cause the name change from Source to AppCatalog to be particularly confusing.
+
+# 5 API Details
+
+```c# (but really MIDL3)
+namespace Microsoft.Management.Deployment
+{
+    /// State of the install.
+    enum InstallProgressState

```

Comment:
Is this maybe `AppInstallProgressState` ? It's not `PackageInstallProgressState`, of course.